modules/server/src/topics/missiontopic.cpp

```diff
@@ -0,0 +1,124 @@
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/server/include/topics/missiontopic.h>
+
+#include <modules/server/include/connection.h>
+#include <modules/server/include/jsonconverters.h>
+#include <openspace/engine/globals.h>
+#include <openspace/mission/missionmanager.h>
+#include <openspace/util/time.h>
+#include <ghoul/logging/logmanager.h>
+#include <modules/spacecraftinstruments/util/imagesequencer.h>
+#include <openspace/util/spicemanager.h>
+
+using nlohmann::json;
+
+namespace openspace {
+
+bool MissionTopic::isDone() const {
+    return true;
+}
+
+nlohmann::json MissionTopic::missionJson() const {
+    const std::map<std::string, Mission>& missions =
+        global::missionManager->missionMap();
+
+    ImageSequencer& sequencer = ImageSequencer::ref();
+    const std::vector<double>& captureTimes = sequencer.captureProgression();
+    std::vector<std::string> captureTimesString(captureTimes.size());
+
+    for (int i = 0; i < captureTimes.size(); i++) {
+        const std::string& str = SpiceManager::ref().dateFromEphemerisTime(
+            sequencer.nextCaptureTime(captureTimes[i]),
+            "YYYY-MM-DDTHR:MN:SC"
+        );
+        captureTimesString[i] = str;
+    }
+    json json;
+    for (auto const& [name, mission] : missions) {
+        nlohmann::json missionJson = createPhaseJson(mission);
+        missionJson["capturetimes"] = captureTimesString;
+        json.push_back(missionJson);
+    }
+
+    return json;
+}
+
+nlohmann::json MissionTopic::createPhaseJson(const MissionPhase& phase) const {
+
+    json phases = json::array();
+    for (const MissionPhase& missionPhase : phase.phases()) {
+        json subphaseJson = createPhaseJson(missionPhase);
+        phases.push_back(subphaseJson);
+    }
+    
+    json milestones = json::array();
+    const std::vector<Milestone>& dates = phase.milestones();
+    for (const Milestone& date : dates) {
+        json jsonDate = {
+            { "date", std::string(date.date.ISO8601())},
+            { "name", date.name }
+        };
+        
+        if (date.description.has_value()) {
+            jsonDate["description"] = date.description.value();
```

Comment:
Quicker way to get to the value of an optional:
`jsonDate["description"] = *date.description;`